google/cloud/aiplatform/experimental/vertex_model/serializers/dataloaders.py

```diff
@@ -15,35 +15,256 @@
 
+def _serialize_remote_dataloader(
+    artifact_uri: str,
+    dataloader_path: str,
+    obj: torch.utils.data.DataLoader,
+    dataset_type: str,
+) -> (str, str):
+    """Serializes DataLoader object to GCS and stores remotely-sourced data in
+       a run-time bucket
 
-def _serialize_local_dataloader():
-    # finds the local source, and copies
-    # data to the user-designated staging bucket
-    raise NotImplementedError
+    Args:
+        artifact_uri (str): the GCS bucket where the serialized object will reside.
+        dataloader_path (str): the path where the origin data used to construct the DataLoader
+                               resides.
+        obj (torch.utils.data.DataLoader): the pytorch DataLoader to serialize.
+        dataset_type (str): the intended use of the dataset (ie. training, testing)
 
+    Returns:
+        The GCS path pointing to the serialized DataLoader, the GCS path pointing to the
+        serialized origin data.
+    """
 
-def _deserialize_local_dataloader():
-    # read the data from user-designated staging bucket and
-    # reformat to a DataLoader
-    raise NotImplementedError
+    # TODO(b/195442091): Check if uri is actually a local path and write to a local
+    #                    location if that is the case.
 
+    # Create a client object
+    client = storage.Client(
+        project=initializer.global_config.project,
+        credentials=initializer.global_config.credentials,
+    )
 
-def _serialize_dataloader():
-    # introspect to determine which method is called
-    raise NotImplementedError
+    # Retrieve the source and blob names from the dataloader path
+    (
+        source_bucket_name,
+        source_blob_prefix,
+    ) = utils.extract_bucket_and_prefix_from_gcs_path(dataloader_path)
 
+    # Retrieve the source and blob names from the artifact URI
+    (
+        destination_bucket_name,
+        destination_blob_prefix,
+    ) = utils.extract_bucket_and_prefix_from_gcs_path(artifact_uri)
 
-def _deserialize_dataloader():
-    # introspect to determine which method is called
-    raise NotImplementedError
+    # Retrieve the blob and bucket name of the original GCS object
+    dataloader_path = pathlib.Path(dataloader_path)
+    local_file_name = dataset_type + "_" + dataloader_path.name
+    source_blob_name = local_file_name
+
+    if source_blob_prefix:
+        source_blob_name = "/".join([source_blob_prefix, source_blob_name])
+
+    source_bucket = client.bucket(source_bucket_name)
+    source_blob = source_bucket.blob(source_blob_name)
+
+    # Create a bucket and blob using the artifact URI
+    destination_bucket = client.bucket(destination_bucket_name)
+
+    destination_blob_name = source_blob_name
+    if destination_blob_prefix:
+        destination_blob_name = "/".join(
+            [destination_blob_prefix, destination_blob_name]
+        )
+
+    # Copy over the object from the source bucket to the new bucket
+    blob_copy = source_bucket.copy_blob(
```

Comment:
This doesn't seem to be necessary. I think we can assume we can retrieve the data from the original remote source.

Reply:
Yes, we can provide guidance that credentials provided to `aiplatform.init` should have permissions to access remote data sources referenced by objects that are passed to `fit`.

In most cases this should work using Application Default Credentials.